Deflake policy startup issue

liggitt opened the pull request:
Fixes issues with initial `ensure` calls not being allowed because the policy cache wasn't updated yet.

Things like this:
```
F1124 08:45:26.030730   12507 run_components.go:393] Unable to initialize namespaces: unable to refresh the security allocation UID blocks: User "system:openshift-master" cannot list all namespaces in the cluster
```

liggitt commented:
[test]

liggitt commented:
@deads2k

deads2k commented on pkg/cmd/server/origin/ensure.go:
```diff
@@ -177,6 +180,18 @@ func (c *MasterConfig) ensureComponentAuthorizationRules() {
 		}
 
+		// Wait until the policy cache has caught up before continuing
+		err := wait.Poll(100*time.Millisecond, 30*time.Second, func() (done bool, err error) {
```
Wow that's a long interval.

deads2k commented on pkg/cmd/server/origin/ensure.go:
```diff
@@ -177,6 +180,18 @@ func (c *MasterConfig) ensureComponentAuthorizationRules() {
 
+		// Wait until the policy cache has caught up before continuing
+		err := wait.Poll(100*time.Millisecond, 30*time.Second, func() (done bool, err error) {
+			if _, err := c.PolicyClient().ClusterPolicies().List(labels.Everything(), fields.Everything()); err == nil {
```
Check what you mean.  You want to wait until some user has some permission.

deads2k replied:
Hmm... I guess this does an indirect check because the `PolicyClient` isn't allowed to read `ClusterPolicies` unless the authorization cache is up to date.  I'd still like a more straightforward check.

liggitt commented:
addressed comments, also switched to always wait for the cache, even when there was an existing policy, to ensure later initialization steps don't fail because the policy cache wasn't up to date

deads2k commented:
lgtm [merge]

liggitt commented:
flake #6437, [merge]

deads2k commented:
UI https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_origin/4447/

re[merge]
